### PR TITLE
Use `getMultiple()` to fetch associations as well in L2 cache fetch operations

### PR DESCRIPTION
**Pull request by lcobucci**
Just as it was done on #5831

### CODE REVIEW

**Comment by lcobucci** [comment]
btw I know this is a bit messy, so please suggest a nice way to improve this (I'm a bit sleepy right now)

File: lib/Doctrine/ORM/Cache/DefaultQueryCache.php [hunk 1]
```diff
@@ -178,14 +176,20 @@ public function get(QueryCacheKey $key, ResultSetMapping $rsm, array $hints = []
                     continue;
                 }
 
-                $collection  = new PersistentCollection($this->em, $assocMetadata, new ArrayCollection());
+                $generateKeys = function ($id) use ($assocMetadata): EntityCacheKey {
```
Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Cache/DefaultQueryCache.php]: PHP7 only feature!

**Review by guilhermeblanco — DISMISSED** [review]

File: lib/Doctrine/ORM/Cache/DefaultQueryCache.php (hunk 1 above)
Comment (lcobucci) [inline lib/Doctrine/ORM/Cache/DefaultQueryCache.php]: `master` already dropped PHP 5.6

Comment (Ocramius) [inline lib/Doctrine/ORM/Cache/DefaultQueryCache.php]: @guilhermeblanco we dropped PHP 5 support in `master` as well

**Comment by LuffyLyon** [comment]
Hello,

This PR will be merged to the 2.6 version that's right?
Because I did the same work in my project to improve performance, I didn't see this PR before, so II've overriden the DefaultQueryCache class.

The PR is still in improvement?

**Comment by lcobucci** [comment]
@LuffyLyon this is only for 2.6 indeed and @Ocramius wanted to have some tests but we bumped into mocking hell. We still need to decide what to do with it (there're some duplications that I'm not really happy with)

**Comment by LuffyLyon** [comment]
Ok thank you, which duplications are you talking about? Maybe I can contribute to improving this feature

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Pull request merged**